### PR TITLE
Properly Escape URLs

### LINKED ISSUE #106: Some URLs are escaped twice in sitemap
for the Chinese post title, the generated url will be encoded twice in jekyll-sitemap, cause it can not be open correctly. please fix it.

### PR DESCRIPTION
**Pull request by pathawks**
Fixes #106

### CODE REVIEW

File: lib/jekyll/sitemap_filters.rb [hunk 1]
```diff
@@ -0,0 +1,10 @@
+require 'addressable/uri'
+
+module Jekyll
+  module SitemapFilters
+    def normalize_url(input)
+      Addressable::URI.parse(input).normalize.to_s
+    end
+  end
+end
+Liquid::Template.register_filter(Jekyll::SitemapFilters)
```
Comment (pathawks) [inline lib/jekyll/sitemap_filters.rb]: I would rather only include this filter when rendering the sitemap, but nothing that I do seems to make that work 😒

**Comment by benbalter** [comment]
Did you mean to re-namespace things in this PR? It's making the diff a bit tough to see what's really going on.

**Comment by pathawks** [comment]
> Did you mean to re-namespace things in this PR?

Sorry. Since I am adding another module, I thout it would make sense to split it up, but you are right.
I’ll break that into a second PR.

**Comment by pathawks** [comment]
I put the refactor in its own commit. Is that enough for you to isolate it and see what I am doing?

[This](https://github.com/jekyll/jekyll-sitemap/pull/107/commits/dcdb816f2e006a4558c2ea82cb98f4f3f82cb3e6) is the meat of this change.

File: Gemfile [hunk 2]
```diff
@@ -1,6 +1,8 @@
 source "https://rubygems.org"
 gemspec
 
+gem "addressable", "~>2.4.0"
```
Comment (benbalter) [inline Gemfile]: Doesn't this need to be in the Gemspec?

Comment (pathawks) [inline Gemfile]: DOH! 😫

Comment (pathawks) [inline Gemfile]: Fixed 🐹

**Comment by benbalter** [comment]
LGTM. Nice work.

**Comment by pathawks** [comment]
@jekyllbot: merge

**Pull request merged**